packages/alfa-table/src/helpers.ts

```diff
@@ -0,0 +1,51 @@
+import { Element, Namespace, Node } from "@siteimprove/alfa-dom";
+import { clamp } from "@siteimprove/alfa-math";
+import { Predicate } from "@siteimprove/alfa-predicate";
+import { Ok } from "@siteimprove/alfa-result";
+
+import { parseNonNegativeInteger } from "./microsyntaxes";
+
+const { isElement, hasName, hasNamespace } = Element;
+const { and } = Predicate;
+
+/**
+ * Parse a "span" (colspan/rowspan) attribute on table cell according to specs.
+ * @see https://html.spec.whatwg.org/multipage/tables.html#algorithm-for-processing-rows
+ *      Steps 8 and 9.
+ */
+export function parseSpan(
+  element: Element,
+  name: string,
+  min: number,
+  max: number,
+  failed: number
+): number {
+  return element
+    .attribute(name)
+    .map((attribute) =>
+      parseNonNegativeInteger(attribute.value).map((x) => clamp(x, min, max))
+    )
+    .getOr(Ok.of(failed))
+    .getOr(failed);
+}
+
+/**
+ * @see https://html.spec.whatwg.org/multipage/tables.html#empty-cell
+ */
+export function isEmpty(element: Element): boolean {
+  return (
+    element.children().isEmpty() &&
```

Comment:
This will be `true` only if the element contains _no_ child nodes, including text nodes.